### PR TITLE
Broken Integration Alarm System

### LINKED ISSUE #2944: RFC: Alarm System
## Problem
- sometimes integrations fail, which is outside of our control
- sometimes credentials change or people remove permissions
- whenever that happens, Cal.com fails silently, and the solution is often to disconnect and reconnect an integration (i.e. Google Calendar)

## Solution
- an alarm system via Email
- similar to Zapiers alarm system when a Zap fails
- a clear "call-to-action" how to fix the problem

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR alerts users when an integration is broken.

When a video integration is broken
image

When there's an error with the calendar
image

Fixes #2944

**Environment**: Staging(main branch) / Production

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- Broken video credential
	- In your .env file go to the Daily credentials and delete the api key
	- Edit an event type and set the location to Cal Video
	- Create a booking, you should get the broken integration email for video
- Broken calendar credential
	- Connect a Google calendar
	- In the DB go to the google calendar credential, under key delete some characters from the access token
	- Create a booking, you should get the broken integration email for calendars
## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by joeauyeung — COMMENTED** [review]
@zomars I made a change to the emails and I know you just refactored so just wanted some feedback while I'm working on this.

File: packages/core/CalendarManager.ts [hunk 1]
```diff
@@ -141,7 +142,8 @@ export const createEvent = async (
 
   // TODO: Surfice success/error messages coming from apps to improve end user visibility
   const creationResult = calendar
-    ? await calendar.createEvent(calEvent).catch((e) => {
+    ? await calendar.createEvent(calEvent).catch(async (e) => {
+        await sendBrokenIntegrationEmail(calEvent, "calendar");
```
Comment (joeauyeung) [inline packages/core/CalendarManager.ts:146]: Only send email on error

File: apps/web/public/static/locales/es/common.json [hunk 2]
```diff
@@ -902,5 +902,6 @@
   "web_conference": "Conferencia web",
   "requires_confirmation": "Requiere confirmación",
   "add_exchange2013": "Conectar con Exchange 2013 Server",
-  "add_exchange2016": "Conectar con Exchange 2016 Server"
+  "add_exchange2016": "Conectar con Exchange 2016 Server",
+ "broken_video_action": "No pudimos agregar el  enlace de <1>{props.location}</1>a tu evento agendado. Contacta a tus invitados o actualiza tu calendario para agregar los detalles. Puedes  <2>cambiar la ubicación de tu tipo de event </2> o <3>remover la App y agregarla de nuevo.</3>"
```
Comment (zomars) [inline apps/web/public/static/locales/es/common.json:906]:
```suggestion
  "broken_video_action": "No pudimos agregar el  enlace de <1>{props.location}</1>a tu evento agendado. Contacta a tus invitados o actualiza tu calendario para agregar los detalles. Puedes  <2>cambiar la ubicación de tu tipo de event </2> o <3>remover la App y agregarla de nuevo.</3>"
```

Comment (zomars) [reply apps/web/public/static/locales/es/common.json:906]: Prettier will complain otherwise

**Review by zomars — CHANGES_REQUESTED** [review]
I don't think we need the failed event attachment ICS file

File: packages/emails/src/templates/BrokenIntegrationEmail.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  const { t } = props;
+
+  return (
+    <Trans i18nKey="broken_calendar_action" t={t}>
+      We could not update your <span>{props.calendar}</span>.{" "}
+      <a href={`${WEBAPP_URL}/apps/installed`}>
+        Please check your calendar settings or remove and add your calendar again
+      </a>
+    </Trans>
+  );
```
Comment (zomars) [inline packages/emails/src/templates/BrokenIntegrationEmail.tsx:47]: Made some fixes on the `BrokenVideoIntegration` let's see if you can replicate in here [based on the docs](https://react.i18next.com/latest/trans-component)

**Review by zomars — COMMENTED** [review]
Almost there, I'm just trying to make the translations works. Let's see if one more pass can get us there.

File: packages/emails/src/templates/BrokenIntegrationEmail.tsx [hunk 4]
```diff
@@ -15,7 +15,7 @@ export function getEnumKeyByEnumValue(myEnum: any, enumValue: number | string):
 
 const BrokenVideoIntegration = (props: { location: string; eventTypeId?: number | null; t: TFunction }) => {
   return (
-    <Trans i18nKey="broken_video_action" t={props.t} values={{ location: props.location }}>
+    <Trans i18nKey="broken_video_action" t={props.t}>
```
[the remark this answers is not in the document]
Comment (zomars) [reply packages/emails/src/templates/BrokenIntegrationEmail.tsx:18]: The calendar one

File: apps/web/pages/api/email.ts [hunk 5]
```diff
@@ -5,7 +5,8 @@ import { getTranslation } from "@calcom/lib/server/i18n";
 
 const handler = async (req: NextApiRequest, res: NextApiResponse) => {
   if (process.env.NODE_ENV !== "development") return res.write("Only for development purposes"), res.end();
-  const t = await getTranslation("en", "common");
+  const t = await getTranslation("es", "common");
```
Comment (baileypumfleet) [inline apps/web/pages/api/email.ts:8]: Why has this been changed to es?

File: apps/web/public/static/locales/en/common.json [hunk 6]
```diff
@@ -911,5 +911,10 @@
   "add_exchange2016": "Connect Exchange 2016 Server",
   "specific_issue": "Have a specific issue",
   "browse_our_docs": "browse our docs",
-  "attendee_name": "Attendee's name"
+  "broken_video_action": "We could not add the <1>{{location}}</1> meeting link to your scheduled event. Contact your invitees or update your calendar event to add the details. You can either <3> change your location on the event type </3> or <5>removing and adding the app again.</5>",
```
Comment (baileypumfleet) [inline apps/web/public/static/locales/en/common.json:914]:
```suggestion
  "broken_video_action": "We could not add the <1>{{location}}</1> meeting link to your scheduled event. Contact your invitees or update your calendar event to add the details. You can either <3> change your location on the event type </3> or try <5>removing and adding the app again.</5>",
```

File: packages/emails/src/templates/BrokenIntegrationEmail.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      We could not add the <span>{props.location}</span> meeting link to your scheduled event. Contact your
+      invitees or update your calendar event to add the details. You can either&nbsp;
+      <a
+        href={
+          props.eventTypeId ? `${WEBAPP_URL}/event-types/${props.eventTypeId}` : `${WEBAPP_URL}/event-types`
+        }>
+        change your location on the event type
+      </a>
+      &nbsp;or&nbsp;
```
Comment (baileypumfleet) [inline packages/emails/src/templates/BrokenIntegrationEmail.tsx:27]:
```suggestion
      &nbsp;or try&nbsp;
```

**Review by pumfleet — DISMISSED** [review]
Only a couple of really minor things

**Comment by baileypumfleet** [comment]
Also, e2e broken

**Review by zomars — APPROVED** [review]
Tests fixed LGTM

**Pull request merged**